presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -888,6 +890,7 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
         }
         finally {
             tablePrivilegesCache.invalidate(new UserTableKey(Optional.of(grantee), databaseName, tableName, tableOwner));
+            tablePrivilegesCache.invalidate(new UserTableKey(Optional.empty(), databaseName, tableName, tableOwner));
```

Comment:
what about grant and revoke for roles?

Reply:
works, see the test

Reply:
This seems to be the crux of the change. It would be cool @iirekm, if that could be explained in the description - so we do not have guess from the code what the goal is :)

Reply:
what you mean by "explained in description" - a comment?

the reason of this strange bahavior is that sometimes cache.get is called Optional.of(grantee), sometimes with Optional.empty(), for me this is a bigger issue for another task to simplify and refactory this code

Reply:
Thanks.

I meant to update the description of this PR with what you just said. But this comment explains it too.

Strange, though, what does a get without a specific grantee even mean?

Reply:
`listTablePrivileges` is called with `Optional.empty()` for `principal` when we want to access all privileges. It is set when when we want to access specific user privileges.

I think adding a comment to code could be nice here.

Reply:
done